// srvx/dist/adapters/bunny.mjs
import { t as createWaitUntil } from "../_chunks/_utils2.mjs";
import { r as wrapFetch, t as errorPlugin } from "../_chunks/_plugins.mjs";
const FastURL = URL;
const FastResponse = Response;
function serve(options) {
	return new BunnyServer(options);
}
var BunnyServer = class {
	runtime = "bunny";
	options;
	fetch;
	_started = false;
	#wait;
	constructor(options) {
		this.options = {
			...options,
			middleware: [...options.middleware || []]
		};
		for (const plugin of options.plugins || []) plugin(this);
		errorPlugin(this);
		const fetchHandler = wrapFetch(this);
		this.#wait = createWaitUntil();
		this.fetch = (request) => {
			Object.defineProperties(request, {
				waitUntil: { value: this.#wait?.waitUntil },
				runtime: {
					enumerable: true,
					value: { name: "bunny" }
				},
				ip: {
					enumerable: true,
					get() {
						return request.headers.get("x-real-ip");
					}
				}
			});
			return fetchHandler(request);
		};
		if (!options.manual) this.serve();
	}
	serve() {
		if (typeof Bunny !== "undefined" && Bunny.v1?.serve) {
			if (this._started) return;
			this._started = true;
			Bunny.v1.serve(this.fetch);
		} else throw new Error("[srvx] Bunny runtime not detected.");
	}
	ready() {
		return Promise.resolve(this);
	}
	async close() {
		await this.#wait?.wait();
	}
};
export { FastResponse, FastURL, serve };
